In [1]:
import numpy as np
from scipy import stats

# Семинар 5

## Задача 1.

Когда используется критерий Стьюдента, а когда Z –критерий?

### Решение.

- Z критерий используется при известном стандартном отклонении генеральной совокупности и нормальном распределении.
- Критерий Стьюдента используется при неизвестном стандартном отклонении генеральной совокупности, нормальном распределении и равенстве дисперсий.

## Задача 2 (решить вручную)

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.

Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр

оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [2]:
mu = 17
X = 17.5
alpha = 0.05 # Zt = 1.645
D = 4
n = 100

H0: mu = mu0 ( Нулевая гипотеза)
H1: mu > mu0 ( Альтернативная)

In [3]:
Z = (X-mu)/(D**0.5/n**0.5)
Z

2.5


Т.к. табличное значение Z для альфа  0.05 равно 1.645 и оно меньше чем расчетное 2.5,  можно сделать вывод, что альтернативная гипотеза верна.

## Задача 3. (решить вручную)

Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.

Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190.

Известно, что их веса распределены нормально.

Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

In [4]:
data = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
mu = 200
n = 10
alpha = 0.01

H0: mu = mu0 ( Нулевая гипотеза)
H1: mu !== mu0 ( Альтернативная)

In [5]:
# среднее арифметическое выборки
X = sum(data) / len(data)
X

198.5

In [6]:
# наблюдаемое среднеквадратичное отклонение (несмещенное)
std = (sum( (i - X) ** 2 for  i in data ) / (len(data)-1 )) ** 0.5
std

4.453463071962462

In [7]:
t = (X-mu) / (std/n**0.5)
t

-1.0651074037450896

In [8]:
1-0.05/2

0.975

t табличное для p  =  0.975 и двустороннего теста при степени свободы ( df  =  10 - 1) будет  2.262

Т.к. t табличное  = 2.262 и больше расчетного t = -1.065 можно сделать вывод, что  Нулевая гипотеза  верна

## Задача 4 (решить с помощью функции)

Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [9]:
mothers = [172, 177, 158, 170, 178,175, 164, 160, 169, 165]
daughters = [ 173, 175, 162, 174, 175, 168, 155, 170, 160]
alpha = 0.05

In [10]:
a = np.array(mothers)
b = np.array(daughters)
res  = stats.ttest_ind(a,b)
res.pvalue

0.8101467126543231

In [11]:
if res.pvalue > alpha:
    print ("Статистически значимых различий нет")
else:
    print ("Статистически значимые различия есть")

Статистически значимых различий нет
